javascript/javascript2/week3/JS2W3 warmup/js2w3.js

```diff
@@ -0,0 +1,165 @@
+  navigator.geolocation.getCurrentPosition(success);
+});
+
+function success(pos) {
+  var crd = pos.coords;
+  targetParagraf.innerHTML = `This is the latitude : ${crd.latitude}<br>
+                                This is the longitude: ${crd.longitude}`;
+}
+
+/*-----------------------------------------------------------------------------------------------
+Create a function called runAfterDelay. It has two parameters: delay and callback. 
+When called the function should wait delay seconds and then call the provided callback function. 
+Try and call this function with different delays and different callback functions
+-------------------------------------------------------------------------------------------------*/
+function runAfterDelay(delay, callback) {
+  const toMiliseconds = delay * 1000;
+  setTimeout(callback, toMiliseconds);
+}
+
+const logStringInFive = () => {
+  console.log("This is loged with a callback function in 5 seconds");
+};
+
+const logStringInSeven = () => {
+  console.log("This is loged with a callback function in 7 seconds");
+};
+
+runAfterDelay(5, logStringInFive);
+runAfterDelay(7, logStringInSeven);
+
+/*------------------------------------------------------------------------------------------------
+Check if we have double clicked on the page. 
+A double click is defined by two clicks within 0.5 seconds. 
+If a double click has been detected, log out the text: "double click!"
+--------------------------------------------------------------------------------------------------*/
+const checkDbl = window.addEventListener("dblclick", () => {
+  console.log("Double click!");
+});
+
+/*-------------------------------------------------------------------------------------------------
+Create a function called jokeCreator that has three parameters: shouldTellFunnyJoke - boolean, 
+logFunnyJoke - function and logBadJoke - function. If you set shouldTellFunnyJoke to true 
+it should call the logFunnyJoke function that should log out a funny joke. And vice versa.
+---------------------------------------------------------------------------------------------------*/
+function jokeCreator(shouldTellFunnyJoke, logFunnyJoke, logBadJoke) {
+  if (shouldTellFunnyJoke === "true") {
+    logFunnyJoke();
+  } else if (shouldTellFunnyJoke === "false") {
+    logBadJoke();
+  } else {
+    console.log("It's not boolean");
+  }
+}
+
+const funnyJoke = () => {
+  console.log("Before the invention of the wheel… everything was a drag!");
+};
+
+const badJoke = () => {
+  console.log(
+    "Why don't oysters donate to charity? Because they're shellfish."
+  );
+};
+
+jokeCreator("true", funnyJoke, badJoke);
+
+/*-----------------------------------------------------------------------------------------------------
+Function as a variable => Create funtions that are used in these different ways:
+-------------------------------------------------------------------------------------------------------*/
+let array = [
```

Comment:
You need to use const instead of let, since this array is not going to change :)